rezervo/models.py

```diff
@@ -13,20 +13,26 @@
     UniqueConstraint,
 )
 from sqlalchemy.dialects.postgresql import JSONB, UUID
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column
 
-from rezervo.database.base_class import Base
 from rezervo.schemas.community import UserRelationship
 
 
+class Base(DeclarativeBase):
+    pass
+
+
 class User(Base):
     __tablename__ = "users"
 
-    id = Column(UUID(as_uuid=True), primary_key=True, index=True, default=uuid.uuid4)
-    name = Column(String, unique=True, nullable=False)
-    jwt_sub = Column(String, nullable=True)
-    cal_token = Column(String, nullable=False)
-    preferences = Column(JSONB)
-    admin_config = Column(JSONB)
+    id: Mapped[uuid.UUID] = mapped_column(
```

Comment:
We should rely on derived types where possible (https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#mapped-column-derives-the-datatype-and-nullability-from-the-mapped-annotation). That way we don't have to define the type twice, avoiding inconsistencies.

We will need a custom type map to use PostgreSQL types (https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#customizing-the-type-map)

Reply:
Hey that is nice! 💯 Implemented in https://github.com/mathiazom/rezervo/pull/47/commits/5656915c8c42b5cf432202fe941708ee0202f016

Reply:
Nice! I just added a custom `small_integer` so that we don't default to `SmallInteger` for all future uses of `int` (0cfa8120b62d394f2f3e22fd3218074c95b11abb)